File: 프로그래머스/2/42578. 의상/의상.py
memo = {}

def factorial(n):
    if n in memo:
        return memo[n]
    if n == 0 or n == 1:
        return 1
    memo[n] = n * factorial(n - 1)
    return memo[n]

def combination(n, r):
    return factorial(n) // (factorial(n - r) * factorial(r))

def solution(clothes):
    cloth = {}
    answer = 1
    
    for name, cloth_type in clothes:
        if cloth_type in cloth:
            cloth[cloth_type] += 1
        else:
            cloth[cloth_type] = 1
            
    for key in cloth:
        answer *= (cloth[key] + 1)
        
    return answer - 1

File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;

class Solution {
    public int solution(String[][] clothes) {
        Map<String, Integer> comb = new HashMap<>();
        for(String[] c : clothes) {
            String type = c[1];
            comb.put(type, comb.getOrDefault(type, 0) + 1);
        }
        int answer = 1;
        for(int k : comb.values()) {
            answer *= (k + 1);
        }
        
        
        return answer - 1;
    }
}